File: pyfo/depreciated/FOPPL/anglican-compiler-pytorch/output-pytorch/if1d.py
import torch
from torch.autograd import Variable
import pyfo.distributions as dist
from pyfo.utils.interface import interface

class model(interface):
    @classmethod
    def gen_vars(self):
        return ['x30321']

    @classmethod
    def gen_cont_vars(self):
        return []

    @classmethod
    def gen_if_vars(self):
        return []

    @classmethod
    def gen_disc_vars(self):
        return ['x30321']

    # prior samples
    @classmethod
    def gen_prior_samples(self):
        dist30359 = dist.Normal(mean=0, std=1)
        x30321 = dist30359.sample()   #sample
        x30361 = logical_trans( x30321 > 0)
        dist30362 = dist.Normal(mean=1, std=1)
        y30324 = 1
        x30364 = logical_trans( x30321 > 0)
        x30365 = not logical_trans(x30364)
        dist30366 = dist.Normal(mean=-1, std=1)
        y30327 = 1
        state = {}
        for _gv in self.gen_vars():
            state[_gv] = locals()[_gv]
        return state # dictionary

    # compute pdf
    @classmethod
    def gen_pdf(self, state):
        dist30359 = dist.Normal(mean=0, std=1)
        x30321 =  state['x30321']    # get the x from input arg
        p30360 = dist30359.logpdf( x30321) # from prior
        x30361 = logical_trans( x30321 > 0)
        dist30362 = dist.Normal(mean=1, std=1)
        y30324 = 1
        p30363 = dist30362.logpdf( y30324) if x30361 else 0 # from observe with if
        x30364 = logical_trans( x30321 > 0)
        x30365 = not logical_trans(x30364)
        dist30366 = dist.Normal(mean=-1, std=1)
        y30327 = 1
        p30367 = dist30366.logpdf( y30327) if x30365 else 0 # from observe with if
        logp =  p30360 + p30363 + p30367  # total log joint
        return logp # need to m

File: pyfo/unittests/models/ifmodels/if_1d_models/if_1d.clj
(let [x (sample (normal 0 1))]
  (if (> x 0)
    (observe (normal 1 1) 1)
    (observe (normal -1 1) 1))
  x)

;(let [x1 (sample (normal 0 1))
;      x2 (sample (normal 1 2))
;      x3 (+ x1 x2)]
;  (if (> x3 0)
;    (observe (normal 1 1) 1)
;    (if (> x3 2)
;      (observe (normal 1 3)))
;    (observe (normal -1 1) 1)
;;  (let [x3 (sample (normal 0 5))
;;        b  (- x1 x3)]
;;    (if (> b a)
;;      (observe (gamma 2 4) 2)
;;      (observe (beta  3 1)3))
;  x1 x2 x3))
;(let [x1 (sample (normal 0 1))
;      x2 (sample (normal 0 1))
;      cov [[1 0] [0 1]]
;      y [1 1]]
;  (if (> x1 0)
;    (if (> x2 0)
;      (observe (mvn [1 1] cov) y)
;      (observe (mvn [1 -1] cov) y))
;    (if (> x2 0)
;      (observe (mvn [-1 1] cov) y)
;      (observe (mvn [-1 -1] cov) y)))
;  [x1 x2])
